openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1316 @@
+            # Normalize material elements vector
+            sum_values = sum(mat_vector.values())
+            for elm in mat_vector:
+                mat_vector[elm] /= sum_values
+        self.mat_vector = mat_vector
+
+    def _get_material(self, val):
+        """Helper method for getting openmc material from Material instance or
+        material name or id.
+
+        Parameters
+        ----------
+        val : Openmc.Material or str or int representing material name or id
+
+        Returns
+        -------
+        val : openmc.Material
+            Openmc Material
+        """
+        if isinstance(val, Material):
+            check_value("Material", str(val.id), self.burn_mats)
+
+        elif isinstance(val, str):
+            if val.isnumeric():
+                check_value("Material id", val, self.burn_mats)
+                val = [mat for mat in self.model.materials if mat.id == int(val)][0]
+
+            else:
+                check_value(
+                    "Material name",
+                    val,
+                    [mat.name for mat in self.model.materials if mat.depletable],
+                )
+                val = [mat for mat in self.model.materials if mat.name == val][0]
+
+        elif isinstance(val, int):
+            check_value("Material id", str(val), self.burn_mats)
+            val = [mat for mat in self.model.materials if mat.id == val][0]
+
+        else:
+            ValueError(f"Material: {val} is not supported")
+
+        return val
+
+    @abstractmethod
+    def _model_builder(self, param):
+        """Builds the parametric model to be passed to the
+        :meth:`openmc.search.search_for_keff` method.
+        Callable function which builds a model according to a passed
+        parameter. This function must return an openmc.model.Model object.
+
+        Parameters
+        ----------
+        param :
+            Model material function variable
+
+        Returns
+        -------
+        _model :  openmc.model.Model
+            Openmc parametric model
+        """
+
+    def search_for_keff(self, x, step_index):
+        """Perform the criticality search on parametric material variable.
+        The :meth:`openmc.search.search_for_keff` solution is then used to
+        calculate the new material volume and update the atoms concentrations.
+
+        Parameters
+        ----------
+        x : list of numpy.ndarray
+            Total atoms concentrations
+
+        Returns
+        -------
+        x : list of numpy.ndarray
+            Updated total atoms concentrations
+        root : float
+             Search_for_keff returned root value
+        """
+        # Update AtomNumber with new conc vectors x. Materials are also updated
+        # even though they are also re-calculated when running the search_for_kef
+        self._update_materials(x)
+
+        # Solve search_for_keff and find new value
+        root = self._search_for_keff(0)
```

Comment:
A comment would be useful as to why we are passing a value of `0` here. Because we are starting at a point of making no modifications?